### PR TITLE
python3Packages.angr: 9.0.6281 -> 9.0.6790 and dependencies

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream releases.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 119894` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python38Packages.ailment</li>
    <li>python38Packages.angr</li>
    <li>python38Packages.archinfo</li>
    <li>python38Packages.claripy</li>
    <li>python38Packages.cle</li>
    <li>python38Packages.pyvex</li>
    <li>python39Packages.ailment</li>
    <li>python39Packages.angr</li>
    <li>python39Packages.archinfo</li>
    <li>python39Packages.claripy</li>
    <li>python39Packages.cle</li>
    <li>python39Packages.pyvex</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119894` at 7ce92f76 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.ailment</li>
    <li>python38Packages.angr</li>
    <li>python38Packages.cle</li>
    <li>python38Packages.pyvex</li>
    <li>python39Packages.ailment</li>
    <li>python39Packages.angr</li>
    <li>python39Packages.cle</li>
    <li>python39Packages.pyvex</li>
  </ul>
</details>
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.archinfo</li>
    <li>python38Packages.claripy</li>
    <li>python39Packages.archinfo</li>
    <li>python39Packages.claripy</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/ailment/default.nix:27:0:
```
   |
27 |     description = "The angr Intermediate Language";
   | ^
```
</li>
    <li>warning: unused-argument

Unused argument: `pkgs`.
Near pkgs/development/python-modules/angr/default.nix:21:3:
```
   |
21 | , pkgs
   |   ^
```
</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 119894` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python38Packages.ailment</li>
    <li>python38Packages.angr</li>
    <li>python38Packages.archinfo</li>
    <li>python38Packages.claripy</li>
    <li>python38Packages.cle</li>
    <li>python38Packages.pyvex</li>
    <li>python39Packages.ailment</li>
    <li>python39Packages.angr</li>
    <li>python39Packages.archinfo</li>
    <li>python39Packages.claripy</li>
    <li>python39Packages.cle</li>
    <li>python39Packages.pyvex</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 119894` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>python38Packages.ailment</li>
    <li>python38Packages.angr</li>
    <li>python38Packages.archinfo</li>
    <li>python38Packages.claripy</li>
    <li>python38Packages.cle</li>
    <li>python38Packages.pyvex</li>
    <li>python39Packages.ailment</li>
    <li>python39Packages.angr</li>
    <li>python39Packages.archinfo</li>
    <li>python39Packages.claripy</li>
    <li>python39Packages.cle</li>
    <li>python39Packages.pyvex</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyvex/default.nix [hunk 1]
```diff
@@ -28,7 +28,7 @@ buildPythonPackage rec {
 
   postPatch = ''
     substituteInPlace pyvex_c/Makefile \
-      --replace "CC=gcc" "CC=${stdenv.cc.targetPrefix}cc"
+      --replace "CC?=gcc" "CC=${stdenv.cc.targetPrefix}cc"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyvex/default.nix:31]: We could also set the CC env now.

Comment (fabaff) [reply pkgs/development/python-modules/pyvex/default.nix]: Update to use env variable now. Thanks for the feedback.

**Comment by fabaff** [comment]
@ofborg build python38Packages.angr python39Packages.angr

**Pull request merged**